packages/puppeteer-core/src/cdp/LifecycleWatcher.ts

```diff
@@ -227,41 +223,19 @@ export class LifecycleWatcher {
   }
 
   #checkLifecycleComplete(): void {
-    // We expect navigation to commit.
-    if (!checkLifecycle(this.#frame, this.#expectedLifecycle)) {
-      return;
+    for (const event of this.#expectedLifecycle) {
+      if (!this.#frame._lifecycleEvents.has(event)) {
+        return;
+      }
     }
+
     this.#lifecycleDeferred.resolve();
     if (this.#hasSameDocumentNavigation) {
       this.#sameDocumentNavigationDeferred.resolve(undefined);
     }
     if (this.#swapped || this.#frame._loaderId !== this.#initialLoaderId) {
       this.#newDocumentNavigationDeferred.resolve(undefined);
     }
-
-    function checkLifecycle(
-      frame: CdpFrame,
-      expectedLifecycle: ProtocolLifeCycleEvent[]
-    ): boolean {
-      for (const event of expectedLifecycle) {
-        if (!frame._lifecycleEvents.has(event)) {
-          return false;
-        }
-      }
-      // TODO(#1): Its possible we don't need this check
-      // CDP provided the correct order for Loading Events
-      // And NetworkIdle is a global state
-      // Consider removing
-      for (const child of frame.childFrames()) {
```

Comment:
so was it here only for the empty frames? isn't this code also checks for networkidle events which could come in any order of the frames/subframes? Is it definitely not possible that the load or domcontent loader events arrive after the main frame load/domcontentloaded?

Reply:
So for the Lifecycle events they now (after the linked CL) come in order, that's only true for parsing so anything coming from the HTML or Scripts (evaluateOnNewDocument included) will be correct.
As for the `NetworkIdle` I have to double check but I think the state for every frame is track on the same NetworkIdle detector so it's global for the page (comment also suggest that).

Reply:
I am not sure it can be tracked globally with OOPIFs

Reply:
Yeah you may be correct. I don't think we have such test. I will write one here.